cli/emit.rs

```diff
@@ -694,27 +730,31 @@ pub(crate) fn emit(
     if !is_emittable(&module.media_type, include_js) {
       continue;
     }
+    // todo(dsherret): the below code is racy as it's inserting items from
+    // the graph into the cache that will be read from later. We should keep
+    // all this data in memory.
     let needs_reload =
       options.reload && !options.reload_exclusions.contains(&module.specifier);
     let version = get_version(module.source.as_bytes(), &config_bytes);
     let is_valid = cache
-      .get(CacheType::Version, &module.specifier)
-      .map_or(false, |v| {
-        v == get_version(module.source.as_bytes(), &config_bytes)
-      });
+      .get_emit_data(&module.specifier)
+      .map_or(false, |d| d.version_hash == version);
```

Comment:
Here is the more inefficient code (I haven't measured it), but fixing the todo outlined on line 733 and the issue I mentioned in the PR description should make it so that we don't get any performance penalty for an up to date cache.